### PR TITLE
scss at keywords

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

File: src/test/scss/scssCompletion.test.ts [hunk 1]
```diff
@@ -88,6 +88,36 @@ suite('SCSS - Completions', () => {
...
+				{ label: '@error' },
+				{ label: '@if' },
+				{ label: '@for', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@each', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@while', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@mixin', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@include' }
+			]
+		});
+		testCompletionFor(`@for $i from 1 through 3 { .item-#{$i} { width: 2em * $i; } } @|`, {
```
Comment (octref) [inline src/test/scss/scssCompletion.test.ts]: This fixes an undocumented existing issue: suppose you have 
```scss
@for $i from 1 through 3 {
  .item-#{$i} { width: 2em * $i; }
}

@|
```

Won't trigger any top level completion.

File: src/services/languageFacts.ts [hunk 2]
```diff
@@ -783,7 +783,7 @@ let atDirectiveList: IEntry[];
 export function getAtDirectives(): IEntry[] {
 	if (!atDirectiveList) {
 		atDirectiveList = [];
-		for (let i = 0, len = atDirectives.length; i < len; i++) {
+		for (let i = 0; i < atDirectives.length; i++) {
```
Comment (aeschli) [inline src/services/languageFacts.ts]: Why changing this?

File: src/services/scssCompletion.ts [hunk 3]
```diff
@@ -191,5 +246,22 @@ export class SCSSCompletion extends CSSCompletion {
 		return result;
 	}
 
+	public getCompletionForTopLevel(result: CompletionList): CompletionList {
+		SCSSCompletion.scssAtDirectives.forEach(d => {
+			result.items.push(d);
+		});
+		for (let entry of languageFacts.getAtDirectives()) {
```
Comment (aeschli) [inline src/services/scssCompletion.ts]: Better call `super.getCompletionForTopLevel()` here

File: src/services/scssCompletion.ts [hunk 4]
```diff
@@ -191,5 +246,22 @@ export class SCSSCompletion extends CSSCompletion {
 		return result;
 	}
 
+	public getCompletionForTopLevel(result: CompletionList): CompletionList {
+		SCSSCompletion.scssAtDirectives.forEach(d => {
```
Comment (aeschli) [inline src/services/scssCompletion.ts]: result.items.push(...SCSSCompletion.scssAtDirectives)

File: src/test/scss/scssCompletion.test.ts [hunk 5]
```diff
@@ -88,6 +88,36 @@ suite('SCSS - Completions', () => {
...
+				{ label: '@at-root' },
+				{ label: '@debug' },
+				{ label: '@warn' },
+				{ label: '@error' },
+				{ label: '@if' },
+				{ label: '@for', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@each', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@while', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@mixin', insertTextFormat: InsertTextFormat.Snippet },
+				{ label: '@include' }
```
Comment (aeschli) [inline src/test/scss/scssCompletion.test.ts]: @include should only be used inside @mixin

File: src/services/languageFacts.ts (hunk 2 above)
Comment (octref) [reply src/services/languageFacts.ts]: Originally I put `scss.ts` together with `data/browsers.ts`, and wrote a function `getScssAtDirectives` right below and fixed this declaration of one more temporary variable. Later I decided `scssCompletion.ts` is the better place to put it.

**Pull request merged**